### PR TITLE
Try to verify metadata by CA signed certificate

### PR DESCRIPTION
**Pull request by mrvanes**
This PR tries to verify metadata signature based on CA signed signing certificate if verification cert is CA cert.

### CODE REVIEW

**Comment by mrvanes** [comment]
https://github.com/SURFscz/SRAM-deploy/issues/455

File: src/xmlsec/__init__.py [hunk 1]
```diff
@@ -313,6 +314,27 @@ def _verify(t, keyspec, sig_path=".//{%s}Signature" % NS['ds'], drop_signature=F
             this_cert = xmlsec.crypto.from_keyspec(keyspec, signature_element=sig)
             log.debug("key size: {!s} bits".format(this_cert.keysize))
 
+            # Try verification by CA signed signing certificate
```
Comment (leifj) [inline src/xmlsec/__init__.py:317]: This seems to turn on verification by default as soon as there is a signing certificate in metadata. Is this really what we want?

Comment (mrvanes) [reply src/xmlsec/__init__.py:317]: I discussed this with my collegues at SURF and indeed they indicated anyone that has CA=true in BasicConstraints know what she's doing and presumably wanted to CA sign the signing certificate. And the following tests (verify_directly_issued_by) excludes any misconceptions about that?

Comment (mrvanes) [reply src/xmlsec/__init__.py:317]: And no, it's not about the certificate in metadata, only the one supplied as verification cert, as far as I could understand the code paths?

Comment (leifj) [reply src/xmlsec/__init__.py:317]: I guess that argument makes sense

Comment (leifj) [reply src/xmlsec/__init__.py:317]: I would very much like to have a couple of testcases though: one where we test the happy path and one were we stick ca=true in an end-entity certificate but not trigger pkix validation

Comment (mrvanes) [reply src/xmlsec/__init__.py:317]: From some local testing I can at least confirm we will _not_ trigger the CA code path if the end-entity certificate has CA=true (validation disabled) and still have a succesful pyff run.

**Comment by mrvanes** [comment]
Added two tests: the happy flow and the end-entity has CA=true flow, although it's not possible to not trigger the pkix validation in the latter case due to the nature of the verification (we need to supply the self-signed metadata cert, which has CA=true in that case).

**Pull request merged**